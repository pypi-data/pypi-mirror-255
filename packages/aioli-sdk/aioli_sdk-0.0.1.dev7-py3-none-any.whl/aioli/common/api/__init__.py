from aioli.common.api import errors
from aioli.common.api._util import PageOpts
