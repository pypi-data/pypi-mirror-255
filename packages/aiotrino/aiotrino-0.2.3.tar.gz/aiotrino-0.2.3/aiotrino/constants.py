# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from typing import Any, Optional, Text  # NOQA: mypy types

DEFAULT_PORT = 8080
DEFAULT_SOURCE = "aiotrino-python-client"
DEFAULT_CATALOG = None  # type: Optional[Text]
DEFAULT_SCHEMA = None  # type: Optional[Text]
DEFAULT_AUTH = None  # type: Optional[Any]
DEFAULT_MAX_ATTEMPTS = 3
DEFAULT_REQUEST_TIMEOUT = 30.0  # type: float

HTTP = "http"
HTTPS = "https"

URL_STATEMENT_PATH = "/v1/statement"


class TrinoHeaders:
    CATALOG = "X-Trino-Catalog"
    SCHEMA = "X-Trino-Schema"
    SOURCE = "X-Trino-Source"
    USER = "X-Trino-User"
    CLIENT_INFO = "X-Trino-Client-Info"

    SESSION = "X-Trino-Session"
    SET_SESSION = "X-Trino-Set-Session"
    CLEAR_SESSION = "X-Trino-Clear-Session"

    STARTED_TRANSACTION = "X-Trino-Started-Transaction-Id"
    TRANSACTION = "X-Trino-Transaction-Id"

    PREPARED_STATEMENT = 'X-Trino-Prepared-Statement'
    ADDED_PREPARE = 'X-Trino-Added-Prepare'
    DEALLOCATED_PREPARE = 'X-Trino-Deallocated-Prepare'


class PrestoHeaders:
    CATALOG = "X-Presto-Catalog"
    SCHEMA = "X-Presto-Schema"
    SOURCE = "X-Presto-Source"
    USER = "X-Presto-User"
    CLIENT_INFO = "X-Presto-Client-Info"

    SESSION = "X-Presto-Session"
    SET_SESSION = "X-Presto-Set-Session"
    CLEAR_SESSION = "X-Presto-Clear-Session"

    STARTED_TRANSACTION = "X-Presto-Started-Transaction-Id"
    TRANSACTION = "X-Presto-Transaction-Id"

    PREPARED_STATEMENT = 'X-Presto-Prepared-Statement'
    ADDED_PREPARE = 'X-Presto-Added-Prepare'
    DEALLOCATED_PREPARE = 'X-Presto-Deallocated-Prepare'


HEADERS = TrinoHeaders
