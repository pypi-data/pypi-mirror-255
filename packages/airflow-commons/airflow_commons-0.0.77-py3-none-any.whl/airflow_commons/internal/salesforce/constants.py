ASYNC_PATH = "data/v1/async"
SYNC_PATH = "hub/v1"
DATA_PATH = "data/v1"
ASYNC_JOB_RESULT_PATH = "/{request_id}/results"
ASYNC_JOB_STATUS_PATH = "/{request_id}/status"
AUTH_URL = "https://{subdomain}.auth.marketingcloudapis.com/"
DATA_EXTENSION_KEY_PATH = "/dataextensions/key:{key}/rows"
DATA_EVENT_KEY_PATH = "/dataevents/key:{key}/rowset"
CUSTOM_OBJECT_DATA_PATH = "/customobjectdata/key/{key}/rowset"
TOKEN_PATH = "v2/token"
GRANT_TYPE = "client_credentials"
FILTER_QUERY = "$filter={filter}"
