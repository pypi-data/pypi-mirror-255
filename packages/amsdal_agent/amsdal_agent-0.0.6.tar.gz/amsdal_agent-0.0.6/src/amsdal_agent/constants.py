RESTART_ACTION = 'RESTART'
SHUTDOWN_ACTION = 'SHUTDOWN'

ALLOWED_ACTIONS = [
    RESTART_ACTION,
    SHUTDOWN_ACTION,
]
