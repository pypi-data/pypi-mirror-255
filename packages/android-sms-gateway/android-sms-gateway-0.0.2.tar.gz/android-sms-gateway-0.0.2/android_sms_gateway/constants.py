VERSION = "0.0.2"

DEFAULT_URL = "https://sms.capcom.me/api/3rdparty/v1"
