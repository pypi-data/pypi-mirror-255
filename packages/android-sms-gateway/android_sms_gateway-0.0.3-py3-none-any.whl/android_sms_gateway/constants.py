VERSION = "0.0.3"

DEFAULT_URL = "https://sms.capcom.me/api/3rdparty/v1"
