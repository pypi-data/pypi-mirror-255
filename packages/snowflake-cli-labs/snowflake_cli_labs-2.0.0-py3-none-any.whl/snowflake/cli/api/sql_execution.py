from __future__ import annotations

import logging
from contextlib import contextmanager
from functools import cached_property
from io import StringIO
from textwrap import dedent
from typing import Iterable

from click import ClickException
from snowflake.cli.api.cli_global_context import cli_context
from snowflake.connector.cursor import DictCursor, SnowflakeCursor
from snowflake.connector.errors import ProgrammingError


class SqlExecutionMixin:
    def __init__(self):
        pass

    @property
    def _conn(self):
        return cli_context.connection

    @cached_property
    def _log(self):
        return logging.getLogger(__name__)

    def _execute_string(
        self,
        sql_text: str,
        remove_comments: bool = False,
        return_cursors: bool = True,
        cursor_class: SnowflakeCursor = SnowflakeCursor,
        **kwargs,
    ) -> Iterable[SnowflakeCursor]:
        """
        This is a custom implementation of SnowflakeConnection.execute_string that returns generator
        instead of list. In case of executing multiple queries are executed one by one. This mean we can
        access result of previous queries while evaluating next one. For example, we can print the results.
        """
        self._log.debug("Executing %s", sql_text)
        stream = StringIO(sql_text)
        stream_generator = self._conn.execute_stream(
            stream, remove_comments=remove_comments, cursor_class=cursor_class, **kwargs
        )
        return stream_generator if return_cursors else list()

    def _execute_query(self, query: str, **kwargs):
        *_, last_result = self._execute_queries(query, **kwargs)
        return last_result

    def _execute_queries(self, queries: str, **kwargs):
        return list(self._execute_string(dedent(queries), **kwargs))

    @contextmanager
    def use_role(self, new_role: str):
        """
        Switches to a different role for a while, then switches back.
        This is a no-op if the requested role is already active.
        """
        role_result = self._execute_query(
            f"select current_role()", cursor_class=DictCursor
        ).fetchone()
        prev_role = role_result["CURRENT_ROLE()"]
        is_different_role = new_role.lower() != prev_role.lower()
        if is_different_role:
            self._log.debug("Assuming different role: %s", new_role)
            self._execute_query(f"use role {new_role}")
        try:
            yield
        finally:
            if is_different_role:
                self._execute_query(f"use role {prev_role}")

    def _execute_schema_query(self, query: str):
        self.check_database_and_schema()
        return self._execute_query(query)

    def check_database_and_schema(self) -> None:
        database = self._conn.database
        schema = self._conn.schema
        self.check_database_exists(database)
        self.check_schema_exists(database, schema)

    def check_database_exists(self, database: str) -> None:
        if not database:
            raise Exception(
                """
                Database not specified. Please update connection to add `DATABASE` parameter,
                or re-run command using `--dbname` option.
                Use `snow connection list` to list existing connections
                """
            )
        try:
            self._execute_query(f"USE DATABASE {database}")
        except ProgrammingError as e:
            raise Exception(
                f"""
            Exception occurred: {e}. Make sure you have `DATABASE` parameter in connection or `--dbname` option provided
            Use `snow connection list` to list existing connections
            """
            ) from e

    def check_schema_exists(self, database: str, schema: str) -> None:
        if not schema:
            raise Exception(
                """
                Schema not specified. Please update connection to add `SCHEMA` parameter,
                or re-run command using `--schema` option.
                Use `snow connection list` to list existing connections
                """
            )
        try:
            self._execute_query(f"USE {database}.{schema}")
        except ProgrammingError as e:
            raise Exception(
                f"""
            Exception occurred: {e}. Make sure you have `SCHEMA` parameter in connection or `--schema` option provided
            Use `snow connection list` to list existing connections
            """
            ) from e

    def to_fully_qualified_name(self, name: str):
        current_parts = name.split(".")
        if len(current_parts) == 3:
            # already fully qualified name
            return name.upper()

        if not self._conn.database:
            raise ClickException(
                "Default database not specified in connection details."
            )

        if len(current_parts) == 2:
            # we assume this is schema.object
            return f"{self._conn.database}.{name}".upper()

        schema = self._conn.schema or "public"
        return f"{self._conn.database}.{schema}.{name}".upper()
