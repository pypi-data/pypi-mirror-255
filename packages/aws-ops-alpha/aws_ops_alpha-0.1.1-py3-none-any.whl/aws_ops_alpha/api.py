# -*- coding: utf-8 -*-

"""
Usage example:

    >>> import aws_ops_alpha.api as aws_ops_alpha
    >>> aws_ops_alpha.runtime
    ...
"""

# ordered from the module having the least dependencies to the one having most dependencies
from . import constants
from .logger import logger
from .constants import CommonEnvNameEnum
from .constants import EnvVarNameEnum
from .constants import AwsTagNameEnum
from .constants import AwsOpsSemanticBranchEnum
from .env_var import temp_env_var
from .env_var import normalize_env_var_name
from .env_var import get_environment_aws_account_id_in_ci
from .env_var import get_environment_iam_role_arn_in_dev_server
from .env_var import temp_env_var
from .runtime.api import RunTimeGroupEnum
from .runtime.api import RunTimeEnum
from .runtime.api import Runtime
from .runtime.api import runtime
from .multi_env.api import EnvNameValidationError
from .multi_env.api import validate_env_name
from .multi_env.api import BaseEnvNameEnum
from .multi_env.api import env_emoji_mapper
from .multi_env.api import EnvNameEnum
from .multi_env.api import detect_current_env
from .git.api import InvalidSemanticNameError
from .git.api import SemanticBranchRule
from .git.api import GitRepo
from .git.api import extract_semantic_branch_name_for_multi_repo
from .git.api import extract_semantic_branch_name_for_mono_repo
from .git.api import MultiGitRepo
from .git.api import MonoGitRepo
from .boto_ses.api import AbstractBotoSesFactory
from .boto_ses.api import AlphaBotoSesFactory
from .config.api import BaseConfig
from .config.api import BaseEnv
from .config.api import T_BASE_CONFIG
from .config.api import T_BASE_ENV
from .aws_helpers import python_helpers
from .aws_helpers import aws_cdk_helpers
from .aws_helpers import aws_lambda_helpers
from .aws_helpers import aws_ecr_helpers
from .aws_helpers import aws_chalice_helpers

# user may not need the following feature to write their application code
# these features are typically used in CI/CD and devops only
try:
    from .aws_helpers import rich_helpers
except ImportError: # pragma: no cover
    pass

try:
    from .project.api import simple_python_project
    from .project.api import simple_cdk_project
    from .project.api import simple_config_project
    from .project.api import simple_lambda_project
    from .project.api import simple_lbd_container_project
    from .project.api import simple_lbd_agw_chalice_project
except ImportError:  # pragma: no cover
    pass

try:
    from .boostrap import api as boostrap
except ImportError:  # pragma: no cover
    pass
