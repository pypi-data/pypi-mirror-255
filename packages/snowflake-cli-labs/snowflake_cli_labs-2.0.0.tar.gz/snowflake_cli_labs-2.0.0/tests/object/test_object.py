from unittest import mock

import pytest
from snowflake.cli.api.constants import SUPPORTED_OBJECTS


@mock.patch("snowflake.connector.connect")
@pytest.mark.parametrize(
    "object_type, expected",
    [
        ("compute-pool", "compute pools"),
        ("network-rule", "network rules"),
        ("database", "databases"),
        ("function", "functions"),
        # ("job", "jobs"),
        ("procedure", "procedures"),
        ("role", "roles"),
        ("schema", "schemas"),
        ("service", "services"),
        ("secret", "secrets"),
        ("stage", "stages"),
        ("stream", "streams"),
        ("streamlit", "streamlits"),
        ("table", "tables"),
        ("task", "tasks"),
        ("user", "users"),
        ("warehouse", "warehouses"),
        ("view", "views"),
        ("image-repository", "image repositories"),
    ],
)
def test_show(
    mock_connector, object_type, expected, mock_cursor, runner, snapshot, mock_ctx
):
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    result = runner.invoke(["object", "list", object_type], catch_exceptions=False)
    assert result.exit_code == 0, result.output
    assert ctx.get_queries() == [f"show {expected} like '%%'"]


DESCRIBE_TEST_OBJECTS = [
    ("compute-pool", "compute-pool-example"),
    ("network-rule", "network-rule-example"),
    ("integration", "integration"),
    ("network-rule", "network rule"),
    ("database", "database-example"),
    ("function", "function-example"),
    # ("job", "job-example"),
    ("procedure", "procedure-example"),
    ("role", "role-example"),
    ("schema", "schema-example"),
    ("service", "service-example"),
    ("secret", "secret-example"),
    ("stage", "stage-example"),
    ("stream", "stream-example"),
    ("streamlit", "streamlit-example"),
    ("table", "table-example"),
    ("task", "task-example"),
    ("user", "user-example"),
    ("warehouse", "warehouse-example"),
    ("view", "view-example"),
]


@mock.patch("snowflake.connector")
@pytest.mark.parametrize("object_type, object_name", DESCRIBE_TEST_OBJECTS)
def test_describe(
    mock_connector, object_type, object_name, mock_cursor, runner, snapshot
):
    mock_connector.connect.return_value.execute_stream.return_value = (
        None,
        mock_cursor(
            rows=[("ID", "NUMBER(38,0", "COLUMN"), ("NAME", "VARCHAR(100", "COLUMN")],
            columns=["name", "type", "kind"],
        ),
    )
    result = runner.invoke(["object", "describe", object_type, object_name])
    assert result.exit_code == 0, result.output
    assert result.output == snapshot


@mock.patch("snowflake.connector")
def test_describe_fails_image_repository(mock_cursor, runner, snapshot):
    result = runner.invoke(["object", "describe", "image-repository", "test_repo"])
    assert result.exit_code == 1, result.output
    assert result.output == snapshot


DROP_TEST_OBJECTS = [
    *DESCRIBE_TEST_OBJECTS,
    ("image-repository", "image-repository-example"),
]


@mock.patch("snowflake.connector")
@pytest.mark.parametrize(
    "object_type, object_name",
    DROP_TEST_OBJECTS,
)
def test_drop(mock_connector, object_type, object_name, mock_cursor, runner, snapshot):
    mock_connector.connect.return_value.execute_stream.return_value = (
        None,
        mock_cursor(rows=[f"{object_name} successfully dropped."], columns=["status"]),
    )

    result = runner.invoke(["object", "drop", object_type, object_name])
    assert result.exit_code == 0, result.output
    assert result.output == snapshot


@pytest.mark.parametrize("command", ["list", "drop", "describe"])
def test_that_objects_list_is_in_help(command, runner):
    result = runner.invoke(["object", command, "--help"])
    for obj in SUPPORTED_OBJECTS:
        if command == "describe" and obj == "image-repository":
            assert obj not in result.output, f"{obj} should not be in help message"
        else:
            assert obj in result.output, f"{obj} in help message"


@pytest.mark.parametrize(
    "command,expect_argument_exception",
    [
        (["object", "drop"], "OBJECT_TYPE"),
        (["object", "drop", "function"], "OBJECT_NAME"),
        (["object", "list"], "OBJECT_TYPE"),
        (["object", "describe"], "OBJECT_TYPE"),
        (["object", "describe", "function"], "OBJECT_NAME"),
    ],
)
def test_throw_exception_because_of_missing_arguments(
    runner, command, expect_argument_exception
):
    result = runner.invoke(command)
    assert result.exit_code == 2, result.output
    assert result.output.__contains__(
        f"Missing argument '{expect_argument_exception}'."
    )
