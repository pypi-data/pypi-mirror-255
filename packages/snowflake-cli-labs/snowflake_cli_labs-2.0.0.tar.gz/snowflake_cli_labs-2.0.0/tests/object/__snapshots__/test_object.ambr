# serializer version: 1
# name: test_describe[compute-pool-compute-pool-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[database-database-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[function-function-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[integration-integration]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[network-rule-network rule]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[network-rule-network-rule-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[procedure-procedure-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[role-role-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[schema-schema-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[secret-secret-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[service-service-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[stage-stage-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[stream-stream-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[streamlit-streamlit-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[table-table-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[task-task-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[user-user-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[view-view-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe[warehouse-warehouse-example]
  '''
  SELECT A MOCK QUERY
  +-----------------------------+
  | name | type        | kind   |
  |------+-------------+--------|
  | ID   | NUMBER(38,0 | COLUMN |
  | NAME | VARCHAR(100 | COLUMN |
  +-----------------------------+
  
  '''
# ---
# name: test_describe_fails_image_repository
  '''
  ╭─ Error ──────────────────────────────────────────────────────────────────────╮
  │ Describe is currently not supported for object of type image-repository      │
  ╰──────────────────────────────────────────────────────────────────────────────╯
  
  '''
# ---
# name: test_drop[compute-pool-compute-pool-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | c      |
  +--------+
  
  '''
# ---
# name: test_drop[database-database-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | d      |
  +--------+
  
  '''
# ---
# name: test_drop[function-function-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | f      |
  +--------+
  
  '''
# ---
# name: test_drop[image-repository-image-repository-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | i      |
  +--------+
  
  '''
# ---
# name: test_drop[integration-integration]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | i      |
  +--------+
  
  '''
# ---
# name: test_drop[network-rule-network rule]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | n      |
  +--------+
  
  '''
# ---
# name: test_drop[network-rule-network-rule-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | n      |
  +--------+
  
  '''
# ---
# name: test_drop[procedure-procedure-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | p      |
  +--------+
  
  '''
# ---
# name: test_drop[role-role-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | r      |
  +--------+
  
  '''
# ---
# name: test_drop[schema-schema-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | s      |
  +--------+
  
  '''
# ---
# name: test_drop[secret-secret-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | s      |
  +--------+
  
  '''
# ---
# name: test_drop[service-service-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | s      |
  +--------+
  
  '''
# ---
# name: test_drop[stage-stage-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | s      |
  +--------+
  
  '''
# ---
# name: test_drop[stream-stream-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | s      |
  +--------+
  
  '''
# ---
# name: test_drop[streamlit-streamlit-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | s      |
  +--------+
  
  '''
# ---
# name: test_drop[table-table-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | t      |
  +--------+
  
  '''
# ---
# name: test_drop[task-task-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | t      |
  +--------+
  
  '''
# ---
# name: test_drop[user-user-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | u      |
  +--------+
  
  '''
# ---
# name: test_drop[view-view-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | v      |
  +--------+
  
  '''
# ---
# name: test_drop[warehouse-warehouse-example]
  '''
  SELECT A MOCK QUERY
  +--------+
  | status |
  |--------|
  | w      |
  +--------+
  
  '''
# ---
